compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -496,10 +496,36 @@
     }
 
     private void resolveConstant(Env semtypeEnv, Map<String, BLangNode> modTable, BLangConstant constant) {
-        SemType semtype = resolveTypeDefn(semtypeEnv, modTable, constant.associatedTypeDefinition, 0);
+        SemType semtype;
+        if (constant.associatedTypeDefinition != null) {
+            semtype = resolveTypeDefn(semtypeEnv, modTable, constant.associatedTypeDefinition, 0);
+        } else {
+            semtype = evaluateConst(constant);
+        }
         semtypeEnv.addTypeDef(constant.name.value, semtype);
     }
 
+    private SemType evaluateConst(BLangConstant constant) {
+        SemType s;
+        switch (constant.symbol.value.type.getKind()) {
+            case INT:
+                s = SemTypes.intConst((long) constant.symbol.value.value);
```

Comment:
We usually don't try to force a single return, shall we use `return` just for the sake of being consistent with rest of the file.